include/fmt/ranges.h

```diff
@@ -357,42 +357,57 @@ template <typename Char, typename... T>
 struct formatter<tuple_join_view<Char, T...>, Char> {
   template <typename ParseContext>
   FMT_CONSTEXPR auto parse(ParseContext& ctx) -> decltype(ctx.begin()) {
-    return ctx.begin();
+    return do_parse(ctx, std::integral_constant<size_t, sizeof...(T)>{});
```

Comment:
Thinking more of it, why not use `for_each` here and get rid of `do_parse`?

Reply:
`std::for_each` must used on something provides LegacyInputIterator, thus cannot be used on tuple. The original code here was to deal this.

Reply:
We can do use `std::apply` with folded expression since C++17. But the check of return value becomes harder, and not elegant.

https://stackoverflow.com/questions/16387354/template-tuple-calling-a-function-on-each-element

Reply:
OK, I guess `for_each` would also be more cumbersome because we can't use generic lambdas yet.